docs/instructor/autotest.md

```diff
@@ -32,3 +32,22 @@ AutoTest can compute feedback either when a GitHub push event (e.g., a `git push
 * `#force` Admin-user only. Forces the submission to be re-graded (e.g., purges the cached result if it exists and grades it again).
 
 * `#slient` Admin-user only. This is used to invoke the bot, but suppresses feedback. `#silent` is usually used in conjunction with `#force`.
+
+## Avoiding Queue Pile-Ups
+
+AutoTest queues student assignments before they are marked by a grading container. In CPSC courses, it is normal for a large number of students to be enrolled in a course. If a large number of students are enrolled in a course and the students share the same deadline for an assignment, AutoTest may queue a larger number of assignments near the deadline. AutoTest can only concurrently mark a small number of assignments. The queue, therefore, may experience a pile-up of assignments that are waiting to be graded, as assignments leave the queue at a slower rate at which they enter the queue.
+
+When a queue pile-up occurs, AutoTest will continue to function. While AutoTest will continue to function,students must wait for their grade results. This wait may inconvenience students if they urgently need grade feedback to continue with the assignment.
+
+To minimize the potential of a queue pile-up:
+
+1) An AutoGrade container should be optimized to perform grading as quickly as possible.
+  - Any internal container initialization should be done during the container BUILD step when possible.
+  - If possible, cache any upstream dependencies that require a download and installation.
+2) A minimum grade feedback delay should be set that is appropriate for your class.
```

Comment:
Instead of `for your class` here, I would say `for your deliverable` to further clarify that this is a deliverable-by-deliverable constraint.